### PR TITLE
feat: adds support for setting description by http params

### PR DESCRIPTION
**Pull request by tobybatch**
## Description
A clear and concise description of what this pull request adds or changes.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] I verified that my code applies to the guidelines (`composer code-check`)
- [X] I updated the documentation (see [here](https://github.com/kimai/www.kimai.org/tree/master/_documentation))
- [X] I agree that this code is used in Kimai and will be published under the [MIT license](https://github.com/kevinpapst/kimai2/blob/master/LICENSE)

### CODE REVIEW

**Comment by tobybatch** [comment]
@kevinpapst I'd like to add a test for this too, are there tests for the http param stuff? where? and I'll add one for this.

File: src/Controller/TimesheetAbstractController.php [hunk 1]
```diff
@@ -183,6 +183,11 @@ protected function create(Request $request, string $renderTemplate, ProjectRepos
             $entry->setActivity($activity);
         }
 
+        if ($request->query->get('description')) {
+            $description = $activityRepository->find($request->query->get('description'));
```
Comment (kevinpapst) [inline src/Controller/TimesheetAbstractController.php:187]: Copy & Paste

**Review by kevinpapst — CHANGES_REQUESTED** [review]

**Comment by kevinpapst** [comment]
https://github.com/kevinpapst/kimai2/blob/master/tests/Controller/TimesheetControllerTest.php#L279

**Comment by tobybatch** [comment]
And that's why I wanted the pointer to the tests :)

Thanks Kev.

**Comment by tobybatch** [comment]
@kevinpapst I'm getting

```
failed to open stream: Too many open files
```

While running tests, my system limit is high but php seems to be choking it.

**Comment by kevinpapst** [comment]
Either:
- Push and let GitHub actions do the work
- or simply execute only one test with `vendor/bin/phpunit tests/Controller/TimesheetControllerTest.php`

As I dropped SQLite support, please `cp phpunit.xml.dist phpunit.dist` and change to true `<env name="BOOTSTRAP_RESET_DATABASE" value="true"/>`.
You need to create a db and user first, commands can be copy & pasted from here: https://github.com/kevinpapst/kimai2/blob/master/phpunit.xml.dist#L19

**Comment by tobybatch** [comment]
Yeah, I'd got the new SQL test DB settings, that was fine. I was just trying to run the whole test suite.

I'll just run my test. TYhanks

**Comment by tobybatch** [comment]
@kevinpapst Tests added, changes made, linting linted. I've switch it from a draft to a real PR.

**Comment by kevinpapst** [comment]
What do you need that for exactly?
Did you try to inject HTML for a XSS attack?

**Comment by tobybatch** [comment]
Added a test to check html injection can't be done. 

I'm moving my browser extension (https://github.com/tobybatch/kimai-browser-plugin) to use tags rather than a custom method for storing relationship between projects/activities and github tickets. I don't think XSS is a vulnerability as a user must be logged in and the description text is correctly decoded.

**Review by kevinpapst — APPROVED** [review]
I was a bit worried about possible security issues, but there is no real difference between get and post params, so good to merge. Thanks @tobybatch 👍

**Pull request merged**